app/src/org/commcare/activities/CommCareWiFiDirectActivity.java

```diff
@@ -595,21 +599,51 @@ public void prepareFileTransfer() {
             myStatusText.setText(localize("wifi.direct.no.group"));
             return;
         }
+        moveFormRecordsToFiles();
+    }
 
-        zipFiles();
+    private void moveReceivedFiles() {
+        File receiveFolder = new File(toBeSubmittedDirectory);
+        if (!receiveFolder.isDirectory() || !receiveFolder.exists()) {
+            // we haven't received any transfers, just return.
+            return;
+        }
+        // We've received transferred forms. Move them into the to be zipped directory.
+        File[] files = receiveFolder.listFiles();
+        try {
+            Log.d(TAG, "We have " + files.length + " toBeSubmitted files to transfer");
+            for (File instanceDirectory : files) {

```

Comment:
is `instanceDirectory` a directory or a file? Can you confirm it is a directory, and if not, rename to something that denotes a file.